pilot/cmd/pilot-agent/main.go

```diff
@@ -338,8 +361,10 @@ func init() {
 		"Proxy IP address. If not provided uses ${INSTANCE_IP} environment variable.")
 	proxyCmd.PersistentFlags().StringVar(&role.ID, "id", "",
 		"Proxy unique ID. If not provided uses ${POD_NAME}.${POD_NAMESPACE} from environment variables")
-	proxyCmd.PersistentFlags().StringVar(&role.Domain, "domain", "",
+	proxyCmd.PersistentFlags().StringVar(&role.DNSDomain, "domain", "",
 		"DNS domain suffix. If not provided uses ${POD_NAMESPACE}.svc.cluster.local")
+	proxyCmd.PersistentFlags().StringVar(&role.TrustDomain, "trust-domain", "",
+		"The domain to use for identities")
```

Comment:
Can it have dots ? Should it be a URL ( spiffe://foo.com/ns/istio-system/istio-pilot ) ? 

We are still hardcoding istio-pilot and the ns is indirect.

Reply:
I think it can have dots.